src/50_Samples_%26_Templates/Product.html

```diff
@@ -144,21 +149,90 @@
       outline-color: #eb407a;
       background: none;
     }
-    form#order amp-selector:not([disabled]) [option][selected]:not([disabled]) {
+    form#order amp-selector:not([disabled]) [option][selected]:not([disabled]):not(.unavailable), amp-img.selected {
       will-change: box-shadow;
       box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
       transition: box-shadow 0.3s cubic-bezier(.25,.8,.25,1);
       color: #fff;
+    }
+
+    form#order amp-selector:not([disabled]) [option][selected]:not([disabled]):not(.unavailable):not(.square) {
       background-color: #eb407a;
     }
-    form#order amp-selector:not([disabled]) [option]:not([selected]):not([disabled]) {
+
+    form#order amp-selector:not([disabled]) [option][selected]:not([disabled]):not(.unavailable).square {
+      outline: 1px solid #eb407a;
+    }
+
+    form#order amp-selector:not([disabled]) [option]:not([selected]):not([disabled]), amp-img:not([.selected]) {
       will-change: box-shadow;
       box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
       transition: box-shadow 0.3s cubic-bezier(.25,.8,.25,1);
     }
-    form#order amp-selector:not([disabled]) [option]:not([selected]):not([disabled]) amp-img {
-      opacity: 0.5;
+
+    form#order amp-selector[name=color] div:not([selected]):not([disabled]) {
+      opacity: 0.6;
+    }
+    .hide {
+      display: none;
+    }
+    .show {
+      display: block;
     }
+    .unavailable {
+      position: relative;
+      background: gray;
+      overflow: hidden;
+      opacity: 0.6;
+      pointer-events: none;
+      user-select: none;
+    }
+
+.unavailable:before, .unavailable:after {
+    position: absolute;
+    content: '';
+    background: grey;
+    display: block;
+    width: auto;
+    height: 1px;
+    min-width: 56px;
+    -webkit-transform: rotate(-45deg);
+    transform: rotate(-45deg);
+    left: -8px;
+    right: 0;
+    top: 0;
+    bottom: 0;
+    margin: auto;
+}
+
+.unavailable:after {
+    -webkit-transform: rotate(45deg);
+    transform: rotate(45deg);
+}
+
+div.square {
+  width: 40px;
+  height: 40px;
+}
+
+div.square.green {
```

Comment:
nit: it doesn't make sense to combine green with div and square. Green would make sense for other elements as well.